source/operations/install-deploy-manage/upgrade-minio-tenant.rst

```diff
@@ -10,71 +10,107 @@ Upgrade a MinIO Tenant
-This procedure assumes the latest stable Operator version |operator-version-stable|.
+To upgrade a Tenant with ``kubectl``:
 
-.. include:: /includes/k8s/install-minio-kubectl-plugin.rst
+#. In a convenient directory, save the current Tenant configuration to a file using ``kubectl get``:
 
-See :ref:`deploy-operator-kubernetes` for complete documentation on deploying the MinIO Operator.
+    .. code-block:: shell
+       :class: copyable
 
-.. important::
+       kubectl get tenant/my-tenant -n my-tenant-ns -o yaml > my-tenant-base.yaml
 
-   If you are upgrading the MinIO Operator, there may be additional changes to the tenant specs required.
-   Refer to the :ref:`MinIO Operator Upgrade <minio-k8s-upgrade-minio-operator>` for specifics on any changes necessary to the tenant spec.
-   The required changes vary based on the Operator version you are upgrading from and to.
-   
-   If required changes are not made to the tenant before upgrading the Operator, your tenant may not be accessible after the upgrade.
+    Replace ``my-tenant`` and ``my-tenant-ns`` with the name and namespace of the Tenant to upgrade.
+    
+    Edit the file to remove the following lines:
+
+    - ``creationTimestamp:``
+    - ``resourceVersion:``
+    - ``uid:``
+    - ``selfLink:`` (if present)
+
+    For example, remove the highlighted lines:
+
+    .. code-block:: shell
+       :emphasize-lines: 2, 6, 7
+
+       metadata:
+         creationTimestamp: "2024-05-29T21:22:20Z"
+         generation: 1
+         name: my-tenant
+         namespace: my-tenant-ns
+         resourceVersion: "4699"
+         uid: d5b8e468-3bed-4aa3-8ddb-dfe1ee0362da
+
+#. In the same directory, create a ``kustomization.yaml`` file with contents resembling the following:
+
+   .. code-block:: shell
+      :class: copyable
+
+      apiVersion: kustomize.config.k8s.io/v1beta1
+      kind: Kustomization
 
-1) Validate the Active MinIO Version
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      resources:
+      - my-tenant-base.yaml
 
-Use the :mc-cmd:`kubectl minio tenant info` command to return a summary of the MinIO Tenant, including the new Pool:
+      patches:
+      - path: upgrade-minio-tenant.yaml
 
-.. code-block:: shell
-   :class: copyable
+   Replace ``my-tenant-base.yaml`` with the name of the file containing the ``kubectl get`` output from the previous step.
 
-   kubectl minio tenant info TENANT_NAME \
-     --namespace TENANT_NAMESPACE  
 
-- Replace ``TENANT_NAME`` with the name of the Tenant.
-- Replace ``TENANT_NAMESPACE`` with the namespace of the Tenant.
+#. Also in the same directory, create a ``upgrade-minio-tenant.yaml`` file with contents resembling the following:
+
+   .. code-block:: shell
+      :class: copyable
+
+      apiVersion: minio.min.io/v2
+      kind: Tenant
+
+      metadata:
+        name: my-tenant
+        namespace: my-tenant-ns
+
+      spec:
+        image: minio/minio:RELEASE.2024-05-28T17-19-04Z
```

Comment:
Can we use the version tag instead of specifying a version here? This will be a pain to maintain if we don't.

Reply:
Now I remember why that wasn't fixed, I got bit by `:substitutions:` again 😆